src/Core/src/Platform/iOS/MauiWKWebView.cs

```diff
@@ -90,14 +90,23 @@ public void LoadHtml(string? html, string? baseUrl)
 				LoadHtmlString(html, baseUrl == null ? new NSUrl(NSBundle.MainBundle.BundlePath, true) : new NSUrl(baseUrl, true));
 		}
 
-		public void LoadUrl(string? url)
+		public async void LoadUrl(string? url)
 		{
 			try
 			{
 				var uri = new Uri(url ?? string.Empty);
 				var safeHostUri = new Uri($"{uri.Scheme}://{uri.Authority}", UriKind.Absolute);
 				var safeRelativeUri = new Uri($"{uri.PathAndQuery}{uri.Fragment}", UriKind.Relative);
-				NSUrlRequest request = new NSUrlRequest(new NSUrl(new Uri(safeHostUri, safeRelativeUri).AbsoluteUri));
+				var safeFullUri = new Uri(safeHostUri, safeRelativeUri);
+				NSUrlRequest request = new NSUrlRequest(new NSUrl(safeFullUri.AbsoluteUri));
+
+				_ = _handler.TryGetTarget(out var handler);
+
+				if ((handler?.HasCookiesToLoad(safeFullUri.AbsoluteUri) ?? false) && !(OperatingSystem.IsIOSVersionAtLeast(11) || OperatingSystem.IsTvOSVersionAtLeast(11)))
+					return;
+
+				if (handler is not null)
+					await handler.SyncPlatformCookiesAsync(safeFullUri.AbsoluteUri);
```

Comment:
Why ignoring the value of the TryGetTarget method and then need to check on the nullability of the out variable? The code will look a lot nicer.
```suggestion
				var safeFullUri = new Uri(safeHostUri, safeRelativeUri);
				NSUrlRequest request = new NSUrlRequest(new NSUrl(safeFullUri.AbsoluteUri));

				if (_handler.TryGetTarget(out var handler)) {
				    if (handler.HasCookiesToLoad(safeFullUri.AbsoluteUri) && !(OperatingSystem.IsIOSVersionAtLeast(11) || OperatingSystem.IsTvOSVersionAtLeast(11))) 
				        return;
				    await handler.SyncPlatformCookiesAsync(safeFullUri.AbsoluteUri);
				} 
```
If nullability if giving you issues we should fix the TryGetTarget method to use https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.notnullwhenattribute?view=net-7.0 and that way the compiler won't give you issues.